### PR TITLE
d2: init at 0.0.13

### PR DESCRIPTION
**Pull request by dit7ya**
######### Description of changes

A modern diagram scripting language that turns text to diagrams
https://d2lang.com

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/d2/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  vendorSha256 = "sha256-/BEl4UqOL4Ux7I2eubNH2YGGl4DxntpI5WN9ggvYu80=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X oss.terrastruct.com/d2/lib/version.Version=${version}"
+  ];
+
+  subPackages = [ "cmd/d2" ];
```
[the remark this answers is not in the document]
Comment (dit7ya) [reply pkgs/tools/text/d2/default.nix:30]: That is the default plugin and is already included in the binary.

File: pkgs/tools/text/d2/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
```
[the remark this answers is not in the document]
Comment (dit7ya) [reply pkgs/tools/text/d2/default.nix:31]: Thanks. Pushed an update.

**Comment by aaronjheng** [comment]
Result of `nixpkgs-review pr 202355` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>d2</li>
  </ul>
</details>

**Review by aaronjheng — APPROVED** [review]

File: pkgs/tools/text/d2/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ldflags = [
+    "-s"
+    "-w"
+    "-X oss.terrastruct.com/d2/lib/version.Version=${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = "installManPage ci/release/template/man/d2.1";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/d2/default.nix:28]:
```suggestion
  postInstall = ''
    installManPage ci/release/template/man/d2.1
  '';
```

**Pull request merged**